src/Pester.Runtime.ps1

```diff
@@ -2624,3 +2626,34 @@ function Invoke-InNewScriptScope ([ScriptBlock] $ScriptBlock, $SessionState) {
 
     . $wrapper $Path $Data
 }
+
+function Add-MissingContainerParameters ($RootBlock, $Container, $CallingFunction) {
+    # Adds default values for container parameters not provided by the user.
+    # Also adds real parameter name as variable in Run-phase when alias was used, just like normal PowerShell will.
+
+    # Using AST to get parameter-names as $PSCmdLet.MyInvocation.MyCommand only works for advanced functions/scripts/cmdlets.
+    # No need to filter on parameter sets OR whether default values are set because Powershell adds all parameters (not aliases) as variables
+    # with default value or $null if not specified (probably to avoid error caused by inheritance).
+    $Ast = switch ($Container.Type) {
+        "ScriptBlock" { $container.Item.Ast }
+        "File" {
+            $externalScriptInfo = $CallingFunction.SessionState.InvokeCommand.GetCommand($Container.Item.PSPath, [System.Management.Automation.CommandTypes]::ExternalScript)
+            $externalScriptInfo.ScriptBlock.Ast
+        }
+        default { throw [System.ArgumentOutOfRangeException]"" }
+    }
+
+    if ($null -ne $Ast -and $null -ne $Ast.ParamBlock -and $Ast.ParamBlock.Parameters.Count -gt 0) {
+        $parametersToCheck = foreach ($param in $Ast.ParamBlock.Parameters) { $param.Name.VariablePath.UserPath }
+
+        foreach ($param in $parametersToCheck) {
+            $v = $CallingFunction.SessionState.PSVariable.Get($param)
+            if ((-not $RootBlock.Data.ContainsKey($param)) -and $v) {
+                if ($PesterPreference.Debug.WriteDebugMessages.Value) {
+                    Write-PesterDebugMessage -Scope Discovery "Container parameter '$param' is undefined, adding to container Data with default value $(Format-Nicely $v.Value)."
```

Comment:
Change to Runtime? Not sure which category/scope is correct here

Reply:
yup

Reply:
fixed